src/routes/(app)/c/private/[channel_id]/+page.server.ts

```diff
@@ -0,0 +1,32 @@
+import { getDb } from '$lib/server';
+import { getChannelById } from '$lib/server/services/channels';
+import { getPosts } from '$lib/server/services/content';
+import { error } from '@sveltejs/kit';
+import type { PageServerLoad } from './$types';
+import { canViewChannel } from '$lib/server/services/channels';
+import { assertAuth } from '$lib/server/auth';
+
+export const load: PageServerLoad = (async (event) => {
+    const db = await getDb();
+    const { channel_id } = event.params;
+    const data = await getChannelById(db, channel_id);
+    const channel_name = data.name;
+
+    assertAuth(event);
+    if (!(await canViewChannel(db, event.locals.user.id, data.id))) {
+        return error(401);
```

Comment:
A message would be nice here too. Don't give the user any more information than they had when they requested though. In this case, they asked for a private channel of a given ID. If they don't have access to such a channel, an error message like the following should be appropriate:
`You do not have access to channel with id ${channelId}`